src/selectors/prevThought.ts

```diff
@@ -0,0 +1,76 @@
+import Path from '../@types/Path'
+import State from '../@types/State'
+import expandThoughts from '../selectors/expandThoughts'
+import { getAllChildrenAsThoughts, isVisible } from '../selectors/getChildren'
+import prevSibling from '../selectors/prevSibling'
+import rootedParentOf from '../selectors/rootedParentOf'
+import appendToPath from '../util/appendToPath'
+import hashPath from '../util/hashPath'
+import head from '../util/head'
+import isRoot from '../util/isRoot'
+import parentOf from '../util/parentOf'
+
+/** Checks if a thought is expanded using the expandThoughts selector. */
+const isExpanded = (state: State, path: Path): boolean => {
+  const expandedPathMap = expandThoughts(state, state.cursor)
+  return !!expandedPathMap[hashPath(path)]
+}
+
+/** Gets the last visible descendant of a thought. */
+const lastVisibleDescendant = (state: State, path: Path): Path => {
+  const id = head(path)
+
+  // Only traverse children if the thought or its parent is expanded
+  if (!isExpanded(state, path)) return path
+
+  // When showHiddenThoughts is true, show all children, otherwise filter hidden ones
+  const children = state.showHiddenThoughts
+    ? getAllChildrenAsThoughts(state, id)
+    : getAllChildrenAsThoughts(state, id).filter(child => isVisible(state, child))
```

Comment:
You can use `getChildren`, which gets only visible thoughts by default.

```diff
diff --git a/src/selectors/prevThought.ts b/src/selectors/prevThought.ts
index 5cc7d01a73..7a9f1fed29 100644
--- a/src/selectors/prevThought.ts
+++ b/src/selectors/prevThought.ts
@@ -1,7 +1,7 @@
 import Path from '../@types/Path'
 import State from '../@types/State'
 import expandThoughts from '../selectors/expandThoughts'
-import { getAllChildrenAsThoughts, isVisible } from '../selectors/getChildren'
+import getChildren, { isVisible } from '../selectors/getChildren'
 import prevSibling from '../selectors/prevSibling'
 import rootedParentOf from '../selectors/rootedParentOf'
 import appendToPath from '../util/appendToPath'
@@ -24,9 +24,7 @@ const lastVisibleDescendant = (state: State, path: Path): Path => {
   if (!isExpanded(state, path)) return path
 
   // When showHiddenThoughts is true, show all children, otherwise filter hidden ones
-  const children = state.showHiddenThoughts
-    ? getAllChildrenAsThoughts(state, id)
-    : getAllChildrenAsThoughts(state, id).filter(child => isVisible(state, child))
+  const children = getChildren(state, id)
 
   // If no children, return current path
   if (children.length === 0) return path
```